src/server.tsx

```diff
@@ -58,31 +60,58 @@ app.get('/healthcheck', (req: express.Request, res: express.Response) => {
 interface Epic {
     html: string;
     css: string;
+    meta: EpicTestTracking;
 }
 
 const fiveMinutes = 60 * 5;
 const [, fetchDefaultEpicContentCached] = cacheAsync(fetchDefaultEpicContent, fiveMinutes);
+const [, fetchConfiguredEpicTestsCached] = cacheAsync(fetchConfiguredEpicTests, 60);
 
 const logTargeting = (message: string): void => {
     if (process.env.LOG_TARGETING === 'true') {
         console.log(message);
     }
 };
 
+const findTestAndVariantOrDefault = async (
+    targeting: EpicTargeting,
+): Promise<Result | undefined> => {
+    if (process.env.ENABLE_DYNAMIC_TESTS === 'true') {
+        const tests = await fetchConfiguredEpicTestsCached();
+        return findTestAndVariant(tests, targeting);
+    } else if (shouldNotRenderEpic(targeting)) {
+        return;
+    } else {
+        return fetchDefaultEpicContentCached();
+    }
+};
+
```

Comment:
Hey @nicl this is the code I mentioned on chat. I want the interface here to be consistent whether we're doing full dynamic test logic, or just returning the default epic (i.e. return a `Result`, or rather a `Promise<Result>`). We already had a `Variant` for the default, but not a `Test`. Now I'm creating a stub `Test` in `fetchDefaultEpicContent` so I can have it return a `Result`. But almost all the fields are irrelevant currently because the targeting logic for the default has already happened at this state, the name is the only piece we use (because we return it in the `meta` object.

Reply:
In the case of the default epic, the `meta` we return is slightly odd currently:

```
"meta": {
            "abTestName": "FrontendDotcomRenderingEpic", // makes sense
            "abTestVariant": "remote_epic_test", // we can't know this for the current test since the platform determines the variant!
            "campaignCode": "gdnwb_copts_memco_FrontendDotcomRenderingEpic_remote_epic_test",
            "campaignId": "FrontendDotcomRenderingEpic" // makes sense
        }
```

Reply:
So on the platforms, we could start using the test name returned here straight away instead of hardcoding for tracking events etc. But the variant name doesn't make sense for the current test, which is unusual in that the platform determines the variant, not the service. Maybe not worth worrying about this too much since the current default test will presumably be stopped soon.

Reply:
Hey @tjmw , I've left a general comment that relates to this before I saw this comment, but generally, I wonder if it it might be easier to simply add a separate (new) endpoint to serve the default and then serve variants from the main endpoint. Coercing the default epic ino a full test/Result also seems reasonable - I think either can work - but I suspect separate endpoints might end up simpler.

Reply:
Ah that's an interesting suggestion. The reason I went for a switch in the contributions service is that it feels like that's the authority, and when we switch we'd want that to happen on both platforms at the same time. I'm not _too_ worried about the extra complexity we have at the moment since I imagine once we switch over to serving tests & variants dynamically we'll be able to remove the default path once we're happy the variants are working well. I'll play around with it though. Maybe an alternative would be to branch based on an env var immediately inside the request handler in the contributions service. That way the two code paths would be very separate (similar to having separate end points) but we could still control with an env var here. What do you think?

Reply:
Hey @nicl I went ahead and added the conditional check to the `post('/epic')` handler (d4454d71bdc15486d06cec08852e5a392b5c0dfa) so the split is now much higher level. I think it's an improvement but keeps control of the behaviour in the service. Interested to hear what you think!